lib/sentry/client.ex

```diff
@@ -354,10 +354,27 @@ defmodule Sentry.Client do
     end
   end
 
+  defp update_if_present(map, :stacktrace, fun) do
+    case Map.pop(map, :stacktrace) do
+      {nil, _} ->
+        map
+
+      {value, map} ->
+        if is_nil(value.frames) || value.frames == [] do
+          Map.delete(map, :stacktrace)
+        else
+          Map.put(map, :stacktrace, fun.(value))
+        end
+    end
+  end
```

Comment:
Let's not special-case `update_if_present/3` to handle `:stacktrace`. In general, that leads to somewhat-confusing code because then function behaves differently based on the key, so it's effectively two functions.



You can do this instead:

```elixir
case Map.from_struct(thread) do
  %{stacktrace: %{frames: [_ | _]} = stacktrace} ->
    %{stacktrace | frames: Enum.map(frames, &Map.from_struct/1)}

  thread_as_map ->
    Map.delete(thread_as_map, :stacktrace)
end
```

This is special-cased to handle `:stacktrace` and the `:frames` logic specifically, reads pretty clear IMO.